### ISSUE TITLE
MIDI router adding lots of MIDI channel 16 events

### ISSUE THREAD

**Issue by riban-bw** [zynthian/zynthian-issue-tracking#1043]
# Describe the issue
When I record a simple MIDI melody with MIDI recorder it is adding lots of ghost note on/off messages on MIDI channel 16 (which is the same as the enabled Master Channel). 
# To Reproduce
## Steps to reproduce the behavior:
1. Add Pianoteq to chain 1
2. Add Amsynth to chain 2
3. Start MIDI recorder
4. Select chain 1 and play some notes
5. Stop MIDI recorder
6. Start MIDI playback
 ## Expected behaviour
The recorded notes are replayed.
## Actual behaviour
The recorded notes are replayed and the display flickers wildly as master channel notes are received.
# Additional context
Capturing the output of zynsmf during playback shows:
```
   0: 91 3c 29 note on  (channel  1): pitch  60, velocity  41
   0: 9f 3c 29 note on  (channel 15): pitch  60, velocity  41
   0: 9f 3c 29 note on  (channel 15): pitch  60, velocity  41
   0: 9f 3c 29 note on  (channel 15): pitch  60, velocity  41
   0: 9f 3c 29 note on  (channel 15): pitch  60, velocity  41
   0: 9f 3c 29 note on  (channel 15): pitch  60, velocity  41
   0: 9f 3c 29 note on  (channel 15): pitch  60, velocity  41
   0: 9f 3c 29 note on  (channel 15): pitch  60, velocity  41
   0: 9f 3c 29 note on  (channel 15): pitch  60, velocity  41
   0: 9f 3c 29 note on  (channel 15): pitch  60, velocity  41
   0: 9f 3c 29 note on  (channel 15): pitch  60, velocity  41
   0: 9f 3c 29 note on  (channel 15): pitch  60, velocity  41
   0: 9f 3c 29 note on  (channel 15): pitch  60, velocity  41
   0: 9f 3c 29 note on  (channel 15): pitch  60, velocity  41
   0: 9f 3c 29 note on  (channel 15): pitch  60, velocity  41
   0: 81 3c 40 note off (channel  1): pitch  60, velocity  64
   0: 8f 3c 40 note off (channel 15): pitch  60, velocity  64
   0: 8f 3c 40 note off (channel 15): pitch  60, velocity  64
   0: 8f 3c 40 note off (channel 15): pitch  60, velocity  64
   0: 8f 3c 40 note off (channel 15): pitch  60, velocity  64
   0: 8f 3c 40 note off (channel 15): pitch  60, velocity  64
   0: 8f 3c 40 note off (channel 15): pitch  60, velocity  64
   0: 8f 3c 40 note off (channel 15): pitch  60, velocity  64
   0: 8f 3c 40 note off (channel 15): pitch  60, velocity  64
   0: 8f 3c 40 note off (channel 15): pitch  60, velocity  64
   0: 8f 3c 40 note off (channel 15): pitch  60, velocity  64
   0: 8f 3c 40 note off (channel 15): pitch  60, velocity  64
   0: 8f 3c 40 note off (channel 15): pitch  60, velocity  64
   0: 8f 3c 40 note off (channel 15): pitch  60, velocity  64
   0: 8f 3c 40 note off (channel 15): pitch  60, velocity  64
```
This is 14 ghost events mimicing the real note on/off events but repeated rapidly on channel 16.
Changing MIDI Master Channel does not change the MIDI output. It still creates events on channel 16.
# Configuration
## Hardware
```
Raspberry Pi 4 Model B Rev 1.5
Audio: V5 ADAC
Display: MIPI DSI 800x480 (inverted)
Wiring: V5
I2C: MCP23017@0x20, MCP23017@0x21
Profile: v5
```
## System
```
Debian GNU/Linux 12 (bookworm)
Build Date: 2024-02-01
Memory: 16% (614M/3792M)
SD Card: 95% (26G/29G)
Temperature: 48.2ºC
Overclock: Maximum
```
## MIDI & UI
```
Tuning: 440 Hz
Master Channel: 16
Preload Presets: On
ZS3 (SubSnapShots): On
Power Save: 10 minutes
Audio Levels on Snapshots: On
```
## Software
```
zyncoder: oram (84612a6) 
zynthian-ui: oram (27d8317)
zynthian-sys: oram (7a6c26f)
zynthian-data: oram (db0355a)
zynthian-webconf: oram (5f6f937)
```

**Comment by riban-bw** [comment]
I have checked and these ghost events are being fed into SMF recorder so it is an issue with MIDI router. I have updated the title of this ticket.

**Comment by riban-bw** [comment]
The note on/off events are being mirrored to all ZynMidiRouter ch outs, hence the duplicate (ghost) messages.

**Comment by riban-bw** [comment]
Ah! MIDI router is mirroring the note on/off events to chains that do not exist. So in the example that has two chains, the first two chains behave correctly but the other 14 have these ghost events. Maybe some code that should be filtering out these events is flawed and allowing unconfigured outputs to send events.

**Comment by riban-bw** [comment]
Fixed by commit [5d347132c018094f712f1402bbc4ca73a046b4f9](https://github.com/zynthian/zyncoder/commit/5d347132c018094f712f1402bbc4ca73a046b4f9).

Do not send MIDI messages to chains that do not exist.

**Comment by jofemodo** [comment]
This change breaks "ALL MIDI channels", I would undo right now and look for a better solution.
